lib/validation.js

```diff
@@ -209,43 +209,36 @@ module.exports = function validation (yargs, usage, y18n) {
     return implied
   }
 
+  function coerceValueType (argv, val) {
+    // convert string '1' to number 1
+    let num = Number(val)
+    val = isNaN(num) ? val : num
+
+    if (typeof val === 'number') {
+      // check length of argv._
+      val = argv._.length >= val
+    } else if (val.match(/^--no-.+/)) {
+      // check if key/value doesn't exist
+      val = val.match(/^--no-(.+)/)[1]
+      val = !argv[val]
+    } else {
+      // check if key/value exists
+      val = argv[val]
+    }
+    return val
+  }
+
   self.implications = function implications (argv) {
     const implyFail = []
 
     Object.keys(implied).forEach((key) => {
       const origKey = key
       ;(implied[key] || []).forEach((value) => {
-        let num
         let key = origKey
         const origValue = value
+        // eslint-disable-next-line
+        [key, value] = [key, value].map((item) => coerceValueType(argv, item))
```

Comment:
@bcoe It seems we aren't getting the intended values while destructuring `key` and `value`.

Reply:
The issue disappears if you do not use destructuring, but neither do I understand why, nor am I able to reproduce such a destructuring issue in a simpler code.

```suggestion
        key = coerceValueType(argv, key)
        value = coerceValueType(argv, value)
```

But in the end, IMHO, having 2 lines is more readable than using a map here...

Reply:
And this way you can also drop the "please don't lint my hack" hint above ;-)